smack-omemo/src/main/java/org/jivesoftware/smackx/omemo/util/BareJidEncoder.java

```diff
@@ -0,0 +1,60 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.omemo.util;
```

Comment:
I wonder if this shouldn't be moved somewhere under `org.jivesoftware.smack.util` in smack-core, after all, it is not really OMEMO specific and other parts of Smack could benefit from it as well.